velox/functions/prestosql/aggregates/tests/ChecksumAggregateTest.cpp

```diff
@@ -125,25 +128,29 @@ TEST_F(ChecksumAggregateTest, doubles) {
   assertSingleGroupChecksum<double>({{}}, "h8rrhbF5N54=");
   assertSingleGroupChecksum<double>({99.9}, "iVY+6I1lKyo=");
   assertSingleGroupChecksum<double>({1, 2, 3}, "AACEg9cR14o=");
+  assertSingleGroupChecksum<double>({kNaN, kNaN, kNaN}, "AACMau93L28=");
 
   assertGroupingChecksum<int8_t, double>(
       {'a', 'b', 'a'}, {1, 2, 3}, {"AACEI6XSDyU=", "AAAAYDI/x2U="});
-
   assertGroupingChecksum<int8_t, double>(
       {'a', 'b', 'a', 'a'}, {1, 2, 3, {}}, {"AAAAYDI/x2U=", "h8pvqVZMR8M="});
+  assertGroupingChecksum<int8_t, double>(
+      {1, 1, 2}, {kNaN, kNaN, kNaN}, {"AAAIR0qlH0o=", "AACEI6XSDyU="});
 }
 
 TEST_F(ChecksumAggregateTest, reals) {
   assertSingleGroupChecksum<float>({1}, "/23UDiDdm9A=");
   assertSingleGroupChecksum<float>({{}}, "h8rrhbF5N54=");
   assertSingleGroupChecksum<float>({99.9}, "IX/UyPhj6MY=");
   assertSingleGroupChecksum<float>({1, 2, 3}, "b/j7Q4YtV+g=");
+  assertSingleGroupChecksum<float>({kNaNF, kNaNF, kNaNF}, "AmWPYoutLK0=");
 
   assertGroupingChecksum<int8_t, float>(
       {'a', 'b', 'a'}, {1, 2, 3}, {"Vswv9sY4wxY=", "GSzMTb/0k9E="});
-
   assertGroupingChecksum<int8_t, float>(
       {'a', 'b', 'a', 'a'}, {1, 2, 3, {}}, {"3ZYbfHiy+rQ=", "GSzMTb/0k9E="});
+  assertGroupingChecksum<int8_t, float>(
+      {1, 1, 2}, {kNaNF, kNaNF, kNaNF}, {"rJhf7Fwec3M=", "Vswvdi6PuTk="});
```

Comment:
Can you create a checksum with only +0's and another with -0's , and validate that they are different/same ?

Reply:
Added. Thank you for the suggestion!